prprocessor/__main__.py

```diff
@@ -282,13 +320,65 @@ async def update_redmine_on_issues(pull_request: Mapping, issues: Iterable[Issue
 
 @process_event_actions('pull_request', {'opened', 'ready_for_review', 'reopened', 'synchronize'})
 @process_webhook_payload
-async def on_pr_modified(*, pull_request: Mapping, **other) -> None:  # pylint: disable=unused-argument
-    await run_pull_request_check(pull_request)
+async def on_pr_modified(*, action: str, pull_request: Mapping, **_kw) -> None:
+    commits_valid_style = await run_pull_request_check(pull_request)
+
+    original_labels: set[str] = {label['name'] for label in pull_request['labels']}
+    labels_before = Label.load_labels(original_labels)
+    labels = labels_before.copy()
+
+    if action == 'opened':
+        labels.add(Label.NOT_YET_REVIEWED)
+
+    if action == 'synchronize' and Label.WAITING_ON_CONTRIBUTOR in labels:
+        labels.remove(Label.WAITING_ON_CONTRIBUTOR)
+        if pull_request['review_comments'] > 0:
+            labels.add(Label.NEEDS_RE_REVIEW)
+
+    if not commits_valid_style:
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+
+    # TODO: handle None value (result is being calculated by GH) and resubmit later?
+    if pull_request['mergable'] is False:
+        # TODO: post message the PR has a conflict?
+        labels.remove(Label.NEEDS_RE_REVIEW)
+        labels.remove(Label.NOT_YET_REVIEWED)
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+
+    labels_to_add = labels - labels_before
+    labels_to_remove = labels_before - labels
+
+    # TODO: branch labels
+
+    await update_pr_labels(pull_request, labels_to_add, labels_to_remove)
+
+
+@process_event_actions('pull_request_review', {'submitted'})
+@process_webhook_payload
+async def on_pr_review_assign_labels(*, pull_request: Mapping, review: Mapping, **_kw) -> None:
+    labels_before = Label.load_labels(label['name'] for label in pull_request['labels'])
+    labels = labels_before.copy()
+
+    # TODO: look at review['author_association'] to see if the review has permissions?
```

Comment:
This is not a regression, right? The current code also accepts reviews by everyone and their dogs?

Reply:
Indeed. I noted it in https://github.com/theforeman/prprocessor/pull/177#discussion_r1318915511 but then expanded the TODO to be clearer.

Reply:
ah yeah, and my brain filtered your comment out :D